src/common/lock_manager.h

```diff
@@ -94,8 +96,10 @@ class LockGuard {
 
   LockGuard(LockGuard &&guard) : lock_(guard.lock_) { guard.lock_ = nullptr; }
 
+  LockGuard &operator=(LockGuard &&) = default;
```

Comment:
IMHO it's not a well-formed move assignment operator, which can lead to double unlock.